### PR TITLE
NIAD-2955: Medication resource missing when Medication Context has not been reset

### PR DESCRIPTION
**Pull request by MartinWheelerMT**
## What

Have introduced a new Unit Test and due to DI Scopes, the best way to fix this appears to be with better exception handling for the medication mapper to ensure the Medication context is reset regardless of succe
## Why
Due to an edge case in the medication mapping that occurs when a EhrExtract has failed processing whilst mapping a medication, and that previous extract had already mapped a medication which is also in the next EhrExtract to be processed.
This leads to this medication not been mapped in the final bundle.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated the [Changelog](/CHANGELOG.md) with details of my change in the UNRELEASED section if this change will affect end users

[mapping-docs]: https://github.com/NHSDigital/patient-switching-adaptors-mapping-documentation

### CODE REVIEW

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import org.springframework.stereotype.Service;
 
 import lombok.AllArgsConstructor;
-
+]
```
Comment (ole4ryb) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java:28]: Martin, I think this causes a failure and needs to be removed.

Comment (MartinWheelerMT) [reply gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java:28]: Yes, that has been removed there, that was left over in error when clearing unused exports / testing scoping.

**Review by ole4ryb — APPROVED** [review]

**Comment by adrianclay** [comment]
I wonder if this functionality is better tested at the unit level:
- E2E tests are slow
- If the scenario stops throwing an exception, re-creating this failure within the E2E context will be very difficult.
- Unit tests are good at injecting random exceptions (hopefully)

**Review by adrianclay — APPROVED** [review]
